python/tank/bootstrap/manager.py

```diff
@@ -202,6 +203,28 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
+    def _get_async_bootstrap(self):
+        """
+        Boolean property that specifies how the engine bootstrap logic should be executed.
+
+        By default, when ```async_bootstrap`` is false, the whole engine bootstrap logic
+        will be executed synchronously in the main application thread.
+
+        When ``async_bootstrap` is true, an :class:`sgtk.Sgtk` instance will be bootstrapped asynchronously
+        in a background thread, followed by launching the engine synchronously
+        in the main application thread. This will allow the main application to continue
```

Comment:
The general idea here is to avoid the app store downloads happening in a blocking way - these can take several seconds. The engine init is comparably really quick! I might be wrong with latest version of maya, but i don't think the engine could start up async right now. a lot of the maya api isn't thread safe, there is a ton of global state in maya and because of engine init hooks we don't always know what code is being executed etc.

Like for the context switch, maybe we could introduce a per-engine property later on where each engine can declare if they can start up async or not? Or something like that? What do you think @josh-t ?